src/carousel/Carousel.js

```diff
@@ -0,0 +1,96 @@
+import React, { Component } from 'react';
+import './Carousel.css';
+
+export default class Carousel extends Component {
+  constructor(props) {
+    super(props);
+    this.currentImageIndex = 1;
+    this.transitionDuration = this.props.transitionDuration;
+    this.imagesCount = this.props.imgs.length;
+    this.state = { isSliding: false, isAnimating: false, leftPosition: 0 };
+    this.changePhoto = this.changePhoto.bind(this);
+  }
+  /* */
+  componentDidMount() {
+    this.carouselContainerWidth = this.carouselElement.offsetWidth;
+    this.setState({ leftPosition: -1 * this.carouselContainerWidth });
+  }
+  render() {
+    let { imgs } = this.props;
+    imgs = this.cloneImages(imgs);
+    return (
+      <div className="carousel" ref={ref => (this.carouselElement = ref)}>
+        {this.renderSliderElement(imgs)}
```

Comment:
no need for a function here, you can create a var above and render the images there and then just drop everything in the return between curly braces